### PR TITLE
Develop/bugfix es 5

### PR DESCRIPTION
**Pull request by andrei-svistunou**

### CODE REVIEW

**Comment by andrei-svistunou** [comment]
[Review](https://trello.com/c/nI5Zy9iC/11-review)

File: js/calc.js [hunk 1]
```diff
@@ -1,195 +1,246 @@
...
+  }
 
-  deleteLastSymbol: function () {
+};
 
-    if (!this.appData.currentState.isFinishCalculate) {
+Calculator.prototype.enterNegativeSign = function (value) {
+  var index = this.currentState.enteredOperators.length;
+  var countOfNumerals = this.currentState.enteredNumerals.length;
+  var negativeSign = '-';
```
Comment (hamanovich) [inline js/calc.js]: it looks like constant, so minus sign can be configurable or added to some init object.

File: js/calc.js [hunk 2]
```diff
@@ -1,195 +1,246 @@
...
-        if (this.appData.currentState.enteredNumerals[countOperands - 1] === '') {
-          this.appData.currentState.enteredNumerals.pop();
-        }
+  if (this.currentState.canEnterOperator) {
 
-      } else {
-        this.appData.currentState.enteredOperators.pop();
-      }
+    if (this.currentState.lastSymbol in this.accordanceOperators) {
+      this.bufferOfStates.pop();
```
Comment (hamanovich) [inline js/calc.js]: pop() method is hard and mutate array, instead of use arr.slice(0, -1) here and any below

File: js/calc.js [hunk 3]
```diff
@@ -205,62 +256,57 @@ CalcDataService.prototype.utils = {
...
+      if (operators[i] && operators[i].importance === 2) {
+        result = this[operators[i].name](result, values[++j]);
 
-  if (value) {
-    this.currentState.currentExpression += value;
-  } else {
-    this.currentState.currentExpression = this.currentState.currentExpression.slice(0, -1);
-  }
+        if (operators[j] && operators[j].importance === 2) {
+          continue;
```
Comment (hamanovich) [inline js/calc.js]: continue should be removed.

**Pull request merged**